frontend/src/lib/k8s/apiProxy.ts

```diff
@@ -540,7 +540,7 @@ export async function streamResult(
       socket = stream(watchUrl, x => cb(x.object), { isJson: true });
     } catch (err) {
       console.error('Error in api request', { err, url });
-      if (errCb) errCb(err as ApiError, cancel);
+      if (errCb && typeof errCb === 'function') errCb(err as ApiError, cancel);
```

Comment:
The `StreamErrCb` is a function, so I am surprised we need this check. Shouldn't Ts catch this?

Reply:
It's getting `{}` for some reason when it crashes. But I couldn't understand the code well enough to track it down.

Also, I think that errCb is optional, despite what the type says. Because other functions that take errCb are optional. I guess there are some `any` in there and that improving the typing would perhaps find the issue.

Reply:
Please add a `// @todo` comment for us to track down the real cause of this and we then merge this if it fixes the issue.

Reply:
Added the todo comment.